src/test/compile-fail/feature-gate-fn_must_use.rs

```diff
@@ -0,0 +1,20 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// gate-test-fn_must_use
+
+#[must_use]
+fn use_it() -> bool {
```

Comment:
The test is written wrongly, it should add an `//~ ERROR #[must_use] on function is experimental` to one of these two lines (depends on which span is used to report the error).